torch/nn/utils/stateless.py

```diff
@@ -1,6 +1,5 @@
-import warnings
 import contextlib
-from typing import Any, Callable, Dict, Iterator, List, Tuple, Union
+from typing import Any, Callable, Dict, Iterator, List, Tuple, Union, Set, Optional
```

Comment:
Let's stack this change on top of https://github.com/pytorch/pytorch/pull/89213 so we have the full picture in mind. Did you want to update torch.func.functional_call in this PR or a follow-up?